builtin/providers/aws/resource_aws_elasticache_replication_group.go

```diff
@@ -0,0 +1,231 @@
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cache_node_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"automatic_failover": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"num_cache_clusters": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+			"parameter_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"engine": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "redis",
+			},
+			"engine_version": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsElasticacheReplicationGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	var hasParameterGroup bool
+
+	replicationGroupId := d.Get("replication_group_id").(string)
+	description := d.Get("description").(string)
+	cacheNodeType := d.Get("cache_node_type").(string)
+	automaticFailover := d.Get("automatic_failover").(bool)
+	numCacheClusters := d.Get("num_cache_clusters").(int)
+	parameterGroupName, hasParameterGroup := d.GetOk("parameter_group_name")
+	engine := d.Get("engine").(string)
+	engineVersion := d.Get("engine_version").(string)
+
+	req := &elasticache.CreateReplicationGroupInput{
+		ReplicationGroupID:		aws.String(replicationGroupId),
+		ReplicationGroupDescription:	aws.String(description),
+		CacheNodeType:            	aws.String(cacheNodeType),
+		AutomaticFailoverEnabled: 	aws.Boolean(automaticFailover),
+		NumCacheClusters:         	aws.Long(int64(numCacheClusters)),
+		Engine:				aws.String(engine),
+		EngineVersion:			aws.String(engineVersion),
+	}
+
+	if hasParameterGroup {
+		req.CacheParameterGroupName = aws.String(parameterGroupName.(string))
+	}
+
+	_, err := conn.CreateReplicationGroup(req)
+	if err != nil {
+		return fmt.Errorf("Error creating Elasticache replication group: %s", err)
+	}
+
+	pending := []string{"creating"}
+	stateConf := &resource.StateChangeConf{
+		Pending:    pending,
+		Target:     "available",
+		Refresh:    ReplicationGroupStateRefreshFunc(conn, d.Id(), "available", pending),
+		Timeout:    15 * time.Minute,
+		Delay:      20 * time.Second,
+		MinTimeout: 5 * time.Second,
+	}
+
+	log.Printf("[DEBUG] Waiting for state to become available: %v", d.Id())
+	_, sterr := stateConf.WaitForState()
+	if sterr != nil {
+		return fmt.Errorf("Error waiting for elasticache (%s) to be created: %s", d.Id(), sterr)
+	}
+
+	d.SetId(replicationGroupId)
+
+	return nil
+}
+
+func resourceAwsElasticacheReplicationGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+	req := &elasticache.DescribeReplicationGroupsInput{
+		ReplicationGroupID: aws.String(d.Id()),
+	}
+
+	res, err := conn.DescribeReplicationGroups(req)
+	if err != nil {
```

Comment:
Can we check the error code here for `ReplicationGroupNotFound` and set the id to `""` if it's not found? See `aws_subnet` for an example:

- https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_subnet.go#L101